src/App.js

```diff
@@ -1,65 +1,91 @@
+import { getProductsFromServer } from "./utils";
 import { Routes, Route } from "react-router-dom";
-import { PrivateRoute } from "./Authentication";
-import { Profile } from "./Authentication";
-import { SignUp } from "./Authentication/Sign-up";
+
+import {
+  ProductListing,
+  Cart,
+  Wishlist,
+  Nav,
+  Home,
+  Footer,
+  ForgotPasswordPage,
+  Login,
+  PrivateRoute,
+  Profile,
+  SignUp,
+  ProductDetailPage,
+  ErrorPage
+} from "./Components";
 
 export default function App() {
   const { dispatch } = useStateContext();
+  const { isUserLoggedIn, userId } = useAuthentication();
 
   useEffect(() => {
     (async () => {
       try {
-        const { response } = await serverRequest({
-          url: "api/products",
-          requestType: "GET"
-        });
-        dispatch({ type: "SET_PRODUCTS", payload: response.data.products });
-      } catch {}
-      try {
-        const { response } = await serverRequest({
-          url: "api/carts",
-          requestType: "GET"
-        });
-        dispatch({ type: "SET_CART", payload: response.data.carts });
-      } catch {}
-      try {
-        const { response } = await serverRequest({
-          url: "api/wishlists",
-          requestType: "GET"
-        });
-
-        dispatch({
-          type: "SET_WISHLIST",
-          payload: response.data.wishlists
-        });
-      } catch {}
+        const {
+          data: { response }
+        } = await getProductsFromServer(
+          "https://uandistoreapi.herokuapp.com/products"
+        );
+        dispatch({ type: "SET_PRODUCTS", payload: response });
+      } catch (error) {
+        console.log(error);
+      }
     })();
   }, []);
 
+  useEffect(() => {
+    if (isUserLoggedIn) {
+      (async () => {
+        try {
+          const {
+            data: { response }
+          } = await getProductsFromServer(
+            `https://uandistoreapi.herokuapp.com/carts/${userId}/cart`
+          );
+
+          dispatch({ type: "SET_CART", payload: response });
+        } catch (error) {
+          console.log(error);
+        }
+
+        try {
+          const {
+            data: { response }
+          } = await getProductsFromServer(
+            `https://uandistoreapi.herokuapp.com/wishlists/${userId}/wishlist`
+          );
+
+          dispatch({ type: "SET_WISHLIST", payload: response });
+        } catch (error) {
+          console.log(error);
+        }
+      })();
+    }
+  }, [isUserLoggedIn]);
+
   return (
     <div className="App">
       <div className="App-container">
         <Nav />
         <Routes>
           <Route path="/" element={<Home />} />
           <Route path="shop" element={<ProductListing />} />
+          <Route path="/shop/:id" element={<ProductDetailPage />} />
+
           <PrivateRoute path="wishlist" element={<Wishlist />} />
           <PrivateRoute path="cart" element={<Cart />} />
           <Route path="/login" element={<Login />} />
```

Comment:
try putting / in front of path or maintain a consistency. Either apply / in all or in no route

Reply:
Done